core/src/main/java/com/cloud/storage/template/OVAProcessor.java

```diff
@@ -246,7 +246,17 @@ public long getTemplateVirtualSize(String templatePath, String templateName) thr
             NodeList diskElements = new OVFHelper().getElementsByTagNameAndPrefix(ovfDoc, "Disk", "ovf");
             for (int i = 0; i < diskElements.getLength(); i++) {
                 Element disk = (Element)diskElements.item(i);
-                long diskSize = Long.parseLong(disk.getAttribute("ovf:capacity"));
+                String diskSizeValue = disk.getAttribute("ovf:capacity");
+                long diskSize = 1;
+                try {
+                    diskSize = Long.parseLong(diskSizeValue);
+                } catch (NumberFormatException e) {
+                    // ASSUMEably the diskSize contains a property for replacement
+                    LOGGER.warn(String.format("the disksize for disk %s is not a valid number: %s", disk.getAttribute("diskId"), diskSizeValue));
+                    // TODO parse the property to get any value can not be done at registration time
+                    //  and will have to be done at deploytime, so for orchestration purposes
+                    //  we now assume, a value of one
```

Comment:
more like a design brainstorm than a TODO

Reply:
@DaanHoogland any specific reason for completely not supporting those kinds of templates instead of assuming the size to be 1 ? I think this approach will have some effect on the usage records of the volume also.

Reply:
@harikrishna-patnala this is only the template and there will be only a "real" disksize on/after deployment. This will only fix registration btw, and assumes deployment is going on as usual, probably for bad reason.